src/System.Console/src/System/ConsoleKeyInfo.cs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+ 
+namespace System
+{
+    public partial struct ConsoleKeyInfo
+    {
+        private char _keyChar;
+        private ConsoleKey _key;
+        private ConsoleModifiers _mods;
+
+        public ConsoleKeyInfo(char keyChar, ConsoleKey key, bool shift, bool alt, bool control)
+        {
+            // Limit ConsoleKey values to 0 to 255, but don't check whether the
+            // key is a valid value in our ConsoleKey enum.  There are a few 
+            // values in that enum that we didn't define, and reserved keys 
+            // that might start showing up on keyboards in a few years.
+            if (((int)key) < 0 || ((int)key) > 255)
+            {
+                throw new ArgumentOutOfRangeException("key", SR.ArgumentOutOfRange_ConsoleKey);
```

Comment:
I realize some things in this change are hard to test, but we should have tests for the things we can verify, e.g. that we're throwing where we expect to be throwing.